docker/release/config/opensearch/opensearch-docker-entrypoint.sh

```diff
@@ -90,22 +67,74 @@ function terminateProcesses {
+    if [[ "$(id -u)" == "0" ]]; then
+        echo "OpenSearch cannot run as root. Please start your container as another user."
+        exit 1
+    fi
+
+    # Parse Docker env vars to customize OpenSearch
+    #
+    # e.g. Setting the env var cluster.name=testcluster
+    # will cause OpenSearch to be invoked with -Ecluster.name=testcluster
+    local opensearch_opts=()
+    while IFS='=' read -r envvar_key envvar_value
+    do
+        # OpenSearch settings need to have at least two dot separated lowercase
+        # words, e.g. `cluster.name`, except for `processors` which we handle
+        # specially
+        if [[ "$envvar_key" =~ ^[a-z0-9_]+\.[a-z0-9_]+ || "$envvar_key" == "processors" ]]; then
+            if [[ ! -z $envvar_value ]]; then
+            opensearch_opt="-E${envvar_key}=${envvar_value}"
+            opensearch_opts+=("${opensearch_opt}")
+            fi
+        fi
+    done < <(env)
+
+    setupSecurityPlugin
+
+    # Enable job control so we receive SIGCHLD when a child process terminates
+    set -m
 
-# Make sure we terminate the child processes in the event of us received TERM (e.g. "docker container stop"), INT (e.g. ctrl-C), EXIT (this script terminates for an unexpected reason), or CHLD (one of the processes terminated unexpectedly)
-trap terminateProcesses TERM INT EXIT CHLD
+    # Make sure we terminate the child processes in the event of us received TERM (e.g. "docker container stop"), INT (e.g. ctrl-C), EXIT (this script terminates for an unexpected reason), or CHLD (one of the processes terminated unexpectedly)
+    trap terminateProcesses TERM INT EXIT CHLD
 
-# Start opensearch
-$OPENSEARCH_HOME/bin/opensearch "${opensearch_opts[@]}" &
-OPENSEARCH_PID=$!
+    # Start opensearch
+    "$@" "${opensearch_opts[@]}" &
+    OPENSEARCH_PID=$!
 
-# Start performance analyzer agent
-$OPENSEARCH_HOME/bin/performance-analyzer-agent-cli > $OPENSEARCH_HOME/logs/performance-analyzer.log 2>&1 &
-PA_PID=$!
+    # Start performance analyzer agent
+    performance-analyzer-agent-cli > $OPENSEARCH_HOME/logs/performance-analyzer.log 2>&1 &
+    PA_PID=$!
 
-# Wait for the child processes to terminate
-wait $OPENSEARCH_PID
-echo "OpenSearch exited with code $?"
-wait $PA_PID
-echo "Performance analyzer exited with code $?"
+    # Wait for the child processes to terminate
+    wait $OPENSEARCH_PID
+    local opensearch_exit_code=$?
+    echo "OpenSearch exited with code ${opensearch_exit_code}"
+
+    wait $PA_PID
+    echo "Performance analyzer exited with code $?"
+
+    return ${opensearch_exit_code}
+}
+
+# Prepend "opensearch" command if no argument was provided or if the first
+# argument looks like a flag (i.e. starts with a dash).
+if [ $# -eq 0 ] || [ "${1:0:1}" = '-' ]; then
+    set -- opensearch "$@"
+fi
+
+if [ "$1" = "opensearch" ]; then
+    # If the first argument is opensearch, then run the setup script.
+    runOpensearch "$@"
+    # Opensearch exit code should be forwarded, but it would be a breaking change.
```

Comment:
Where do you want to forward the return code?

Reply:
I just mean that the script should exit with the same exit code as the opensearch command (which implies that the line that just follows should be uncommented). I rephrased that.

Reply:
Got it.